api/api/admin/media_report.py

```diff
@@ -441,69 +442,79 @@ def decision_create_view(self, request, object_id):
-                decision.moderator = request.user
-                decision.save()
-
-                logger.info(
-                    "Decision created",
-                    decision=decision.id,
-                    action=decision.action,
-                    notes=decision.notes,
-                    moderator=request.user.get_username(),
-                )
+        if request.method != "POST":
+            return redir
 
-                through = through_model.objects.create(
-                    decision=decision,
-                    media_obj=media_obj,
-                )
-                logger.info(
-                    "Through model created",
-                    through=through.id,
-                    decision=decision.id,
-                    media_obj=media_obj.id,
-                )
+        if not request.user.has_perm(f"api.add_{self.media_type}decision"):
+            messages.error(
+                request,
+                "You do not have permission to create decisions.",
+            )
+            return redir
 
-                reports = form.cleaned_data["reports"]
-                count = reports.update(decision=decision)
-                logger.info(
-                    "Decision recorded in reports",
-                    report_count=count,
-                    decision=decision.id,
+        media_obj = self.get_object(request, object_id)
+
+        through_model = {
+            "image": ImageDecisionThrough,
+            "audio": AudioDecisionThrough,
+        }[self.media_type]
+        form = get_decision_form(self.media_type)(request.POST)
+        if form.is_valid():
+            decision = form.save(commit=False)
+            decision.moderator = request.user
+            decision.save()
+
+            logger.debug(
+                "Decision created",
+                decision=decision.id,
+                action=decision.action,
+                notes=decision.notes,
+                moderator=request.user.get_username(),
+            )
+
+            through = through_model.objects.create(
+                decision=decision,
+                media_obj=media_obj,
+            )
+            logger.debug(
+                "Through model created",
+                through=through.id,
+                decision=decision.id,
+                media_obj=media_obj.id,
+            )
+
+            reports = form.cleaned_data["reports"]
+            count = reports.update(decision=decision)
+            logger.debug(
+                "Decision recorded in reports",
+                report_count=count,
+                decision=decision.id,
+            )
+
+            if decision.action in {
+                DecisionAction.DEINDEXED_COPYRIGHT,
+                DecisionAction.DEINDEXED_SENSITIVE,
+            }:
+                messages.debug(
+                    request,
+                    "The media object has been deindexed from ES and deleted from DB.",
```

Comment:
It would be more accurate to say it's the deleted from the media table, as deleted records are kept in a separate table.

```suggestion
                    f"The media object has been deindexed from ES and deleted from the {self.media_type} table.",
```

Reply:
That's true, but technically all the data is gone. Only the UUID of the media item is left in the other table.